src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTokenExtensions.cs

```diff
@@ -13,11 +13,12 @@ namespace Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery
 {
     internal static partial class SyntaxTokenExtensions
     {
-        public static bool IsUsingOrExternKeyword(this SyntaxToken token)
+        public static bool IsStartOfUsingOrExternDirective(this SyntaxToken token)
```

Comment:
hrmm.. this isn't quite accurate now.  prior it was really just askign about the keywords.  now it strongly implies that it's at the start of the directive.  but that's not really true for the checks below anymore.  for exampple `using var x` or `public extern void Foo()` would both return true here, but wouldn't really be the start of a using/extern directive.  Consider keeping the method the same, and possibly adding the global check to the caller?